engines/twine/actor.cpp

```diff
@@ -0,0 +1,543 @@
+			localActor->boudingBox.Y.bottomLeft = bottomLeftY;
+			localActor->boudingBox.Y.topRight = topRightY;
+			localActor->boudingBox.Z.bottomLeft = bottomLeftZ;
+			localActor->boudingBox.Z.topRight = topRightZ;
+		}
+
+		if (currentIndex == -1)
+			return;
+
+		if (localActor->previousAnimIdx == -1)
+			return;
+
+		_engine->_renderer->copyActorInternAnim(bodyTable[currentIndex], bodyTable[localActor->entity]);
+
+		return;
+	}
+
+	localActor->body = -1;
+	localActor->entity = -1;
+
+	localActor->boudingBox.X.bottomLeft = 0;
+	localActor->boudingBox.X.topRight = 0;
+	localActor->boudingBox.Y.bottomLeft = 0;
+	localActor->boudingBox.Y.topRight = 0;
+	localActor->boudingBox.Z.bottomLeft = 0;
+	localActor->boudingBox.Z.topRight = 0;
+}
+
+/** Initialize actors
+	@param actorIdx actor index to init */
+void Actor::initActor(int16 actorIdx) {
+	ActorStruct *actor = &_engine->_scene->sceneActors[actorIdx];
+
+	if (actor->staticFlags.bIsSpriteActor) { // if sprite actor
+		if (actor->strengthOfHit != 0) {
+			actor->dynamicFlags.bIsHitting = 1;
+		}
+
+		actor->entity = -1;
+
+		initSpriteActor(actorIdx);
+
+		_engine->_movements->setActorAngleSafe(0, 0, 0, &actor->move);
+
+		if (actor->staticFlags.bUsesClipping) {
+			actor->lastX = actor->X;
+			actor->lastY = actor->Y;
+			actor->lastZ = actor->Z;
+		}
+
+	} else {
+		actor->entity = -1;
+
+		initModelActor(actor->body, actorIdx);
+
+		actor->previousAnimIdx = -1;
+		actor->animType = 0;
+
+		if (actor->entity != -1) {
+			_engine->_animations->initAnim(actor->anim, 0, 255, actorIdx);
+		}
+
+		_engine->_movements->setActorAngleSafe(actor->angle, actor->angle, 0, &actor->move);
+	}
+
+	actor->positionInMoveScript = -1;
+	actor->labelIdx = -1;
+	actor->positionInLifeScript = 0;
+}
+
+/** Reset actor
+	@param actorIdx actor index to init */
+void Actor::resetActor(int16 actorIdx) {
+	ActorStruct *actor = &_engine->_scene->sceneActors[actorIdx];
+
+	actor->body = 0;
+	actor->anim = 0;
+	actor->X = 0;
+	actor->Y = -1;
+	actor->Z = 0;
+
+	actor->boudingBox.X.bottomLeft = 0;
+	actor->boudingBox.X.topRight = 0;
+	actor->boudingBox.Y.bottomLeft = 0;
+	actor->boudingBox.Y.topRight = 0;
+	actor->boudingBox.Z.bottomLeft = 0;
+	actor->boudingBox.Z.topRight = 0;
+
+	actor->angle = 0;
+	actor->speed = 40;
+	actor->controlMode = 0;
+
+	actor->info0 = 0;
+	actor->info1 = 0;
+	actor->info2 = 0;
+	actor->info3 = 0;
+
+	actor->brickShape = 0;
+	actor->collision = -1;
+	actor->standOn = -1;
+	actor->zone = -1;
+
+	memset(&actor->staticFlags, 0, 2);
+	memset(&actor->dynamicFlags, 0, 2);
+
+	actor->life = 50;
+	actor->armor = 1;
+	actor->hitBy = -1;
+	actor->lastRotationAngle = 0;
+	actor->lastX = 0;
+	actor->lastY = 0;
+	actor->lastZ = 0;
+	actor->entity = -1;
+	actor->previousAnimIdx = -1;
+	actor->animType = 0;
+	actor->animPosition = 0;
+
+	_engine->_movements->setActorAngleSafe(0, 0, 0, &actor->move);
+
+	actor->positionInMoveScript = -1;
+	actor->positionInLifeScript = 0;
+}
+
+/** Process hit actor
```

Comment:
Method comments should be kept just in the header files rather than in the cpp files

Reply:
Should be fixed - thanks